documentation/specs/proposed/General_perf_onepager.md

```diff
@@ -0,0 +1,42 @@
+# General performance improvements
+MSBuild is the main tool used for building .NET projects. It is used by Visual Studio, the .NET CLI, and other build tools. The performance of MSBuild is critical to the productivity of developers. This document outlines our goals to improve overall performance of MSBuild.
```

Comment:
I wouldn't exclude "C++" nor "SDK projects" in this where that overlaps and falls into the MSBuild "core bucket".

Thanks for this, one more thing I would add would be to ensure that we have:

1) The right telemetry in place to avoid regressions in the wild. We already have this for Visual Studio rebuild, however, we want to fix other scenarios, we should have telemetry in place so that a) we can track that we're making a difference and b) don't regress it.

2) Tests. There's no point fixing performance if someone can make a change very next build and regress it. We have tests in the VS lab, but if we're interested in very specific scenarios, we must have tests running somewhere (our lab can be that if you like) to avoid regressions.

This is how VS does it:

1) Identity scenarios we care about
2) Right tests in lab to track that scenario
3) Have the _right_ telemetry and put on "ShipReady" to monitor and automatically file bugs when it regresses.

Now you folks will be new to a lot of this, and my team can help bring you up to speed how this works and help where needed.

Reply:
I didn't want to exclude any project type :) The .NET was just on top of my mind. I will write it more generic.

I like your enthusiasm :) We will have a working lab tailored for MSBuild to detect regressions. With the VS lab data, we can cover more scenarios, which is great. Looking forward to join forces and speed up the VS.